### PR TITLE
disttask: remove revert subtask

### LINKED ISSUE #50549: disttask: remove reverting subtask
## Enhancement
Revert subtask is used for rollback task. It's created when the task failed or cancelled on each executor node of the task, and they have the same step as the current task step. It has the following issues:
- There's no rollback in any business type now, they are only used to notify us that we need cancel other pending/running subtasks which we can do without Revert subtask.
- Even though we DO need to rollback later, there's no need to create one subtask on each node.
- Node scale in/out, and those revert subtask are not balanced, so they will block the task. And even if they are balanced to active nodes, it's meaningless to run them there.

If rollback is required later, we can:
- Have a special step to do rollback.
- Rollback shouldn't be bind to some node. We can split the rollback into multiple subtasks though.

If we want to do some cleanup(such as cleanup local sorted kv) on each node, we should do it on subtask cleanup. Or start a background routine to do it if cleanup fails, or tidb exit abnormally.

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #50549

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/disttask/framework/storage/task_state.go [hunk 1]
```diff
@@ -59,6 +59,19 @@ func (mgr *TaskManager) FailTask(ctx context.Context, taskID int64, currentState
...
 }
 
+// RevertTask implements the scheduler.TaskManager interface.
+func (mgr *TaskManager) RevertTask(ctx context.Context, taskID int64, taskState proto.TaskState, taskErr error) error {
+	_, err := mgr.ExecuteSQLWithNewSession(ctx, `
+		update mysql.tidb_global_task
+		set state = %?,
+			error = %?,
+			state_update_time = CURRENT_TIMESTAMP()
+		 where id = %? and state = %?`,
```
Comment (ywqzzy) [inline pkg/disttask/framework/storage/task_state.go:69]:
```suggestion
		where id = %? and state = %?`,
```

File: pkg/disttask/framework/testutil/task_util.go [hunk 2]
```diff
@@ -28,12 +28,8 @@ import (
 
 // CreateSubTask adds a new task to subtask table.
 // used for testing.
-func CreateSubTask(t *testing.T, gm *storage.TaskManager, taskID int64, step proto.Step, execID string, meta []byte, tp proto.TaskType, concurrency int, isRevert bool) {
-	state := proto.SubtaskStatePending
-	if isRevert {
-		state = proto.SubtaskStateRevertPending
-	}
-	InsertSubtask(t, gm, taskID, step, execID, meta, state, tp, concurrency)
+func CreateSubTask(t *testing.T, gm *storage.TaskManager, taskID int64, step proto.Step, execID string, meta []byte, tp proto.TaskType, concurrency int) {
```
Comment (ywqzzy) [inline pkg/disttask/framework/testutil/task_util.go:31]: This function is useless now

File: pkg/disttask/framework/proto/task.go [hunk 3]
```diff
@@ -20,40 +20,37 @@ import (
...
-//		   │             │└────────►│reverting├────►│reverted│
-//		   │             ▼          └────┬────┘     └────────┘
-//		   │          ┌──────────┐    ▲  │          ┌─────────────┐
-//		   └─────────►│cancelling├────┘  └─────────►│revert_failed│
-//		              └──────────┘                  └─────────────┘
-//	 1. succeed:		pending -> running -> succeed
-//	 2. failed:			pending -> running -> reverting -> reverted/revert_failed, pending -> failed
-//	 3. canceled:		pending -> running -> cancelling -> reverting -> reverted/revert_failed
-//	 4. pause/resume:	pending -> running -> pausing -> paused -> running
+// Note: if task fail during running, it will end with `reverted` state.
```
Comment (ywqzzy) [inline pkg/disttask/framework/proto/task.go:23]:
```suggestion
// Note: if a task fails during running, it will end with `reverted` state.
```

**Review by ywqzzy — APPROVED** [review]

File: pkg/disttask/framework/testutil/task_util.go (hunk 2 above)
Comment (D3Hunter) [reply pkg/disttask/framework/testutil/task_util.go:31]: a wrapper to create pending subtask

File: pkg/disttask/framework/storage/table_test.go [hunk 4]
```diff
@@ -93,12 +84,10 @@ func TestTaskTable(t *testing.T) {
...
 	require.GreaterOrEqual(t, task4[0].StateUpdateTime, task.StateUpdateTime)
 
-	prevState := task.State
-	task.State = proto.TaskStateRunning
-	retryable, err := gm.UpdateTaskAndAddSubTasks(ctx, task, nil, prevState)
+	err = gm.SwitchTaskStep(ctx, task, proto.TaskStateRunning, proto.StepOne, nil)
 	require.NoError(t, err)
-	require.True(t, retryable)
 
+	task.State = proto.TaskStateRunning
```
Comment (okJiang) [inline pkg/disttask/framework/storage/table_test.go:90]: how about updating state and step in `SwitchTaskStep`, and returning the new task?

**Review by okJiang — COMMENTED** [review]
rest lgtm

File: pkg/disttask/framework/storage/table_test.go (hunk 4 above)
Comment (D3Hunter) [reply pkg/disttask/framework/storage/table_test.go:90]: that's side effects we should avoid

**Review by okJiang — APPROVED** [review]

**Pull request merged**